### PR TITLE
Move some code from core to server

### PR DESCRIPTION
**Pull request by simonbrandhof**
It will help for future refactorings

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java [hunk 1]
```diff
@@ -304,22 +307,6 @@ public boolean plan(DefaultIssue issue, @Nullable ActionPlan actionPlan, IssueCh
...
-
-  public boolean setPastProject(DefaultIssue issue, String previousKey, IssueChangeContext context) {
-    String currentProjectKey = issue.projectKey();
-    issue.setProjectKey(previousKey);
-    return setProject(issue, currentProjectKey, context);
-  }
-
   public boolean setTags(DefaultIssue issue, Collection<String> tags, IssueChangeContext context) {
     Set<String> newTags = Sets.newHashSet(Collections2.transform(
       Collections2.filter(tags, new Predicate<String>() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java]: MAJOR Make this a named "static" inner class. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)

File: sonar-core/src/main/java/org/sonar/core/user/DefaultUser.java [hunk 2]
```diff
@@ -27,8 +27,10 @@
 import org.sonar.api.user.User;
 
 /**
- * @since 3.6
+ * Used only by unit tests. To be replaced by UserDto.
+ * @deprecated
  */
+@Deprecated
 public class DefaultUser implements User {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/user/DefaultUser.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 8 issues:
* MAJOR 6 major
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Make this a named "static" inner class.](https://github.com/SonarSource/sonarqube/blob/1ab936d602273b35cdcf7a883773e3d5527aad19/server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java#L317) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)
* MINOR [Define a constant instead of duplicating this literal "automaticclose" 4 times.](https://github.com/SonarSource/sonarqube/blob/1ab936d602273b35cdcf7a883773e3d5527aad19/server/sonar-server/src/main/java/org/sonar/server/issue/workflow/IssueWorkflow.java#L131) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Make "buildManualTransitions" a "static" method.](https://github.com/SonarSource/sonarqube/blob/1ab936d602273b35cdcf7a883773e3d5527aad19/server/sonar-server/src/main/java/org/sonar/server/issue/workflow/IssueWorkflow.java#L57) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)
* MAJOR [Make "buildAutomaticTransitions" a "static" method.](https://github.com/SonarSource/sonarqube/blob/1ab936d602273b35cdcf7a883773e3d5527aad19/server/sonar-server/src/main/java/org/sonar/server/issue/workflow/IssueWorkflow.java#L128) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)
* MAJOR [The "input" parameter nullability is different in the superclass method, and that should not be changed.](https://github.com/SonarSource/sonarqube/blob/1ab936d602273b35cdcf7a883773e3d5527aad19/server/sonar-server/src/main/java/org/sonar/server/rule/RuleKeyFunctions.java#L44) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)
* MAJOR [The return value nullability of this method is different in the superclass, and that should not be changed.](https://github.com/SonarSource/sonarqube/blob/1ab936d602273b35cdcf7a883773e3d5527aad19/server/sonar-server/src/main/java/org/sonar/server/rule/RuleKeyFunctions.java#L44) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

**Comment by julienlancelot** [comment]
LGTM

**Pull request closed without merging**